config.go

```diff
@@ -161,14 +161,27 @@ func (y *YAML) at(path []string) (interface{}, bool) {
 
 	cur := y.contents
 	for _, segment := range path {
+		// Cast to a mapping type. If this fails, then we ended up on a path
+		// that didn't terminate on a sequence or a scalar.
 		m, ok := cur.(map[interface{}]interface{})
 		if !ok {
 			return nil, false
 		}
-		if _, ok := m[segment]; !ok {
+
+		// Unmarshal the path segement for a comparible key. After all,
+		// YAML scalar types are more than strings (boolean, integer, etc).
```

Comment:
> boolean

ಠ_ಠ

Reply:
You too can have "no: yes" in your YAML for extra confusing fun times.